pdns/dnsdist-lua.cc

```diff
@@ -264,6 +264,40 @@ void checkParameterBound(const std::string& parameter, uint64_t value, size_t ma
   }
 }
 
+static void LuaThread(const std::string code)
+{
+  LuaContext l;
+  // submitToMainThread is camelcased, threadmessage is not.
+  // This follows our tradition of hooks we call being lowercase but functions the user can call being camelcased, but it's up for debate.
+  l.writeFunction("submitToMainThread", [](std::string cmd, std::unordered_map<std::string, std::string> data) {
+    auto lua = g_lua.lock();
+    // maybe offer more than `void`
+    auto func = lua->readVariable<boost::optional<std::function<void(std::string cmd, std::unordered_map<std::string, std::string> data)>>>("threadmessage");
+    if (func) {
+      func.get()(cmd, data);
+    }
+    else {
+      errlog("luathread called submitToMainThread but no threadmessage receiver is defined");
+    }
+  });
+
+  // function threadmessage(data) print("got thread data:") for k,v in pairs(data) do print(k,v) end end
+
+  for (;;) {
+    try {
+      l.executeCode(code);
+      errlog("luaThread exited, restarting in 5 seconds");
```

Comment:
I wonder if it would make sense to make that behaviour optional, so that a Lua thread can actually exit when it's done?

Reply:
I also wondered that!